crates/autopilot/src/auction_transaction.rs

```diff
@@ -0,0 +1,183 @@
+    pub async fn run_forever(self) -> ! {
+        loop {
+            match self.update().await {
+                Ok(true) => (),
+                Ok(false) => tokio::time::sleep(Duration::from_secs(10)).await,
+                Err(err) => {
+                    tracing::error!(?err, "auction transaction update task failed");
+                    tokio::time::sleep(Duration::from_secs(10)).await;
+                }
+            }
+        }
+    }
+
+    /// Update a single settlement event.
+    ///
+    /// Returns whether an update was performed.
+    async fn update(&self) -> Result<bool> {
+        let current_block = self
+            .current_block
+            .borrow()
+            .number
+            .context("no block number")?
+            .as_u64();
+        let reorg_safe_block: u64 = current_block
+            .checked_sub(MAX_REORG_BLOCK_COUNT)
+            .context("no reorg safe block")?;
+        let reorg_safe_block: i64 = reorg_safe_block.try_into().context("convert block")?;
+        let mut ex = self.db.0.acquire().await?;
+        let event = match database::auction_transaction::get_settlement_event_without_tx_info(
+            &mut ex,
+            reorg_safe_block,
+        )
+        .await
+        .context("get_settlement_event_without_tx_info")?
+        {
+            Some(event) => event,
+            None => return Ok(false),
+        };
+        let hash = H256(event.tx_hash.0);
+        tracing::trace!(?hash);
+
+        let transaction = self
+            .web3
+            .eth()
+            .transaction(TransactionId::Hash(hash))
+            .await
+            .with_context(|| format!("get tx {hash:?}"))?
+            .with_context(|| format!("no tx {hash:?}"))?;
+        let from: database::Address = ByteArray(
+            transaction
+                .from
+                .with_context(|| format!("no from {hash:?}"))?
+                .0,
+        );
+        let nonce: i64 = transaction
+            .nonce
+            .try_into()
+            .map_err(|err| anyhow!("{}", err))
+            .with_context(|| format!("convert nonce {hash:?}"))?;
+
+        database::auction_transaction::insert_settlement_tx_info(
+            &mut ex,
+            event.block_number,
+            event.log_index,
+            &from,
+            nonce,
+        )
+        .await
+        .with_context(|| format!("insert_settlement_tx_info {hash:?}"))?;
+
+        // Auctions that were stored before the auction_transaction table existed need to be
+        // inserted into it.
+        // This code can be removed when all old auctions have been processed this way.
+        let auction_id = match database::auction_transaction::get_auction_id_from_tx_hash(
+            &mut ex,
+            &event.tx_hash,
+        )
+        .await
+        .with_context(|| format!("get_auction_id_from_tx_hash {hash:?}"))?
+        {
+            Some(auction_id) => auction_id,
+            None => return Ok(true),
+        };
```

Comment:
This part confuses me a bit...

> Auctions that were stored before the auction_transaction table existed need to be inserted into it.

To me, this means that if `get_auction_id_from_tx_hash` returns `None`, then don't we need to insert something into the `auction_transactions` table instead of exiting?